Core/src/main.cpp

```diff
@@ -43,60 +81,102 @@ int run_glfw() {
 
 	glViewport(0, 0, 800, 600);
 
+	// Call back for changing the viewport size of the OpenGL context.
+	// Context needs to be syncronized to caller thread.
 	glfwSetFramebufferSizeCallback(window, framebuffer_size_callback);
 
+    // primitive shapes
+    Triangle triangle;
+    Square square;
+
+
+	std::vector<System*> systems;
+	Core* core = new Core();
+
+	UI* ui = new UI();
+	Physics* physx = new Physics();
+	Renderer* renderer = new Renderer();
+
+	core->AddSystem(ui);
+	core->AddSystem(physx);
+	core->AddSystem(renderer);
+
+	ui->Initialize();
+	physx->Initialize();
+	renderer->Initialize();
+
+	// Timing
+	double fixedUpdateBuffer = 0.0;
+	double FIXED_UPDATE_INTERVAL = 0.016;
+	auto previousTime = std::chrono::high_resolution_clock::now();
+
+	// We want some check like this visible to the other threads
+	// That way those threads will stop once the window closes. ### Has to be conditional for main thread ###
 	while (!glfwWindowShouldClose(window))
 	{
+		auto currentTime = std::chrono::high_resolution_clock::now();
+
+		std::chrono::duration<double> elapsedTime = currentTime - previousTime;
```

Comment:
rmr to eventually increment fixedUpdateBuffer with this.

```cpp
fixedUpdateBuffer += elapsedTime;
```
or something, later on.